graph/schema.graphqls

```diff
@@ -331,3 +333,51 @@ type DCN {
   """
   vehicle: Vehicle
 }
+
+type EarningNode {
+  """
+  Week reward was issued
+  """
+  week: Int!
+  """
+  Address of Beneficiary that received reward
+  """
+  beneficiary: Address!
+  """
+  Consecutive period of which vehicle was connected
+  """
+  connectionStreak: Int!
```

Comment:
Can you require this? There is a state in which you have device earning information but not streak information.

Reply:
You could consider, in your query, hiding rows that don't have a connection streak set. That would be annoying but it would allow you to be very confident that

1. The streak is there
2. At least one of the devices is populated